Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -767,9 +767,10 @@ extension TaskHandler: ChannelDuplexHandler {
 
         return body.stream(HTTPClient.Body.StreamWriter { part in
             context.eventLoop.assertInEventLoop()
-            return context.writeAndFlush(self.wrapOutboundOut(.body(part))).map {
+            context.write(self.wrapOutboundOut(.body(part))).whenSuccess {
```

Comment:
hmm, this would now _never_ flush any body which means we'll load the whole body into memory before sending.

I think the right thing to do is:

- if you know that whole body straight away (without streaming), then we should do
  `write(.head)`, `.write(.body)`, `.write(end)`, `flush`
- if we actually want to stream the body, then
    `writeAndFlush(.head)`, `writeAndFlush(.body)`, `writeAndFlush(.end)`

Reply:
We also need a test actually that shows that this PR doesn't work. We should have a test which checks that body chunks do arrive at the other end before the next bit is sent out.

Reply:
wrote #219  to test this. This PR should go in after #219 to prove that it works.